kong/db/strategies/init.lua

```diff
@@ -51,31 +51,29 @@ function _M.new(kong_config, strategy, schemas, errors)
       return nil, nil, err
     end
 
-    if Strategy.CUSTOM_STRATEGIES then
-      local custom_strategy = Strategy.CUSTOM_STRATEGIES[schema.name]
-
-      if custom_strategy then
-        local parent_mt = getmetatable(strategy)
-        local mt = {
-          __index = function(t, k)
-            -- explicit parent
-            if k == "super" then
-              return parent_mt
-            end
-
-            -- override
-            local f = custom_strategy[k]
-            if f then
-              return f
-            end
-
-            -- parent fallback
-            return parent_mt[k]
+    local custom_strat = fmt("kong.db.strategies.%s.%s", strategy, schema.name)
+    local pok, mod = pcall(require, custom_strat)
```

Comment:
We have a [load_module_if_exists](https://github.com/Kong/kong/blob/master/kong/tools/utils.lua#L576) utility for this which is more helpful, since it will throw an error when the module fails to load for other reasons than not existing (e.g. syntax errors).

Reply:
good call!